edit/edit.js

```diff
@@ -1284,47 +1300,61 @@ function beautify(event) {
 
 onDOMready().then(init);
 
-function init() {
-  initCodeMirror();
+function createEmptyStyle() {
   const params = getParams();
-  if (!params.id) {
-    // match should be 2 - one for the whole thing, one for the parentheses
-    // This is an add
-    $('#heading').textContent = t('addStyleTitle');
-    const section = {code: ''};
-    for (const i in CssToProperty) {
-      if (params[i]) {
-        section[CssToProperty[i]] = [params[i]];
-      }
+  const style = {
+    id: null,
+    name: '',
+    enabled: true,
+    sections: [{code: ''}]
+  };
+  for (const i in CssToProperty) {
+    if (params[i]) {
+      style.sections[0][CssToProperty[i]] = [params[i]];
     }
-    addSection(null, section);
-    editors[0].setOption('lint', CodeMirror.defaults.lint);
-    editors[0].focus();
-    // default to enabled
-    $('#enabled').checked = true;
-    initHooks();
-    setCleanGlobal();
-    updateTitle();
-    return;
   }
-  // This is an edit
-  $('#heading').textContent = t('editStyleHeading');
-  getStylesSafe({id: params.id}).then(styles => {
-    let style = styles[0];
-    if (!style) {
-      style = {id: null, sections: []};
-      history.replaceState({}, document.title, location.pathname);
+  return style;
+}
+
+function windowLoaded() {
+  if (document.readyState !== 'loading') {
```

Comment:
this code is invoked after onDOMready so readyState will never be `loading`, hence you're resolving immediately instead of waiting for window's load event. You're changing the behavior. Why? I think the old code used window.onload to ensure all scripts finished loading.

Reply:
I don't understand either. Here is the old code:
https://github.com/openstyles/stylus/blob/a8796b0bd2799ad3ac67e3dab663c3d4bbb2a3d0/edit/edit.js#L1321-L1327

Just like you said, onload handler is always executed manually by `window.onload()`. Maybe we should remove `windowLoaded()` entirely since it is useless?

Reply:
Hmm, I guess this was required at some point in the past, but not now that `init` runs after DOMContentLoaded and all lazy-loaded scripts are correctly loaded without specifically waiting for window's load. Yes, remove it and invoke initWithStyle immediately.